extensions/amp-a4a/0.1/test/test-amp-a4a.js

```diff
@@ -709,6 +717,31 @@ describe('amp-a4a', () => {
         expect(a4a.onLayoutMeasure.bind(a4a)).to.throw(/fixed/);
       });
     });
+    it('does not initialize promise chain if display none', () => {
+      xhrMock.onFirstCall().returns(Promise.resolve(mockResponse));
+      return createAdTestingIframePromise().then(fixture => {
+        const doc = fixture.doc;
+        const a4aElement = createA4aElement(doc);
+        a4aElement.style.display = 'none';
+        const a4a = new MockA4AImpl(a4aElement);
+        a4a.onLayoutMeasure();
+        expect(a4a.adPromise_).to.not.be.ok;
+        // Second call w/o display none will cause promise chain to execute.
+        a4aElement.style.display = 'block';
+        a4a.onLayoutMeasure();
+        expect(a4a.adPromise_).to.be.ok;
+      });
+    });
+    it('does not initialize promise chain if height 0', () => {
+      xhrMock.onFirstCall().returns(Promise.resolve(mockResponse));
+      return createAdTestingIframePromise().then(fixture => {
+        const doc = fixture.doc;
+        const a4aElement = createA4aElement(doc, 0);
```

Comment:
Good, but there should probably be a separate test that checks `width == 0` as well.  (Sorry to be a PITA about tests, but it continues to be one of our weak spots.  :-( )